SDK_versioned_docs/version-3.0.0/smart-order-router.md

```diff
@@ -0,0 +1,226 @@
+---
+id: Smart Order Router Integration
+sidebar_position: 3
+---
+
+# Integrating with the Smart Order Router
+The smart order router is a package that exposes a more optimized algorithm for swapping on Uniswap. To use the smart order router, you will create an `AlphaRouter` instance and use the method `route` to get quotes, gas information, and calldata for an optimized swap.
+
+## The Package
+Import `AlphaRouter` from the [smart-order-router](https://www.npmjs.com/package/@uniswap/smart-order-router) package.
+
+```typescript
+import { AlphaRouter } from '@uniswap/smart-order-router'
+```
+
+## The Providers
+
+To create an instance of the AlphaRouter, configure the following providers:
+
+```typescript 
+export type AlphaRouterParams = {
+  chainId: ChainId;
+  provider: providers.BaseProvider;
+  multicall2Provider?: UniswapMulticallProvider;
+  v3SubgraphProvider?: IV3SubgraphProvider;
+  v3PoolProvider?: IV3PoolProvider;
+  v3QuoteProvider?: IV3QuoteProvider;
+  v2SubgraphProvider?: IV2SubgraphProvider;
+  v2PoolProvider?: IV2PoolProvider;
+  v2QuoteProvider?: IV2QuoteProvider;
+  tokenProvider?: ITokenProvider;
+  gasPriceProvider?: IGasPriceProvider;
+  v3GasModelFactory?: IV3GasModelFactory;
+  v2GasModelFactory?: IV2GasModelFactory;
+  blockedTokenListProvider?: ITokenListProvider;
+};
+```
+
+`chainId` [required]
```

Comment:
I considered doing a table, but I think the code blocks are large enough that its pretty hard to read in a table.
I think even putting the default values below is a tough experience because they sometimes help understand what the parameter is. Wondering if there are any other good formats ..